### ISSUE TITLE
Skaffold 1.39.* crashes on build

### ISSUE THREAD

**Issue by akharlamov** [GoogleContainerTools/skaffold#7709]
### Expected behavior
Build succeeds

### Actual behavior
```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.needsRemoteTagging.Tag({{0xc000f7e000, 0x40}, {0xc00090fb00, 0x5b}, {0xc0005ce8c0, 0x47}}, {0x78e1478, 0xc000bd6540}, 0xc00091a780, {0xc000a89620})
	/private/tmp/skaffold-20220625-10976-1ij4m21/pkg/skaffold/build/cache/details.go:94 +0x285
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.(*cache).Build(0xc00091a780, {0x78e13d0, 0xc0006c8280}, {0x788cf60, 0xc000114820}, 0xc000cb6030, {0xc000530ae0, 0x4, 0x4}, {0xc000a89620}, ...)
	/private/tmp/skaffold-20220625-10976-1ij4m21/pkg/skaffold/build/cache/retrieve.go:85 +0x117d
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*Builder).Build(0xc000177080, {0x78e1478, 0xc000a89fb0}, {0x788cf60, 0xc000114820}, {0xc000530ae0, 0x4, 0x4})
	/private/tmp/skaffold-20220625-10976-1ij4m21/pkg/skaffold/runner/build.go:105 +0xa26
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doBuild.func1({0x792f100, 0xc000177080}, {0xc000bb8800, 0x1, 0x1})
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/app/cmd/build.go:72 +0x162
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.withRunner({0x78e13d0, 0xc00029e100}, {0x788cf60, 0xc0005b0000}, 0xc000a23560)
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/app/cmd/runner.go:55 +0x20d
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doBuild({0x78e13d0, 0xc00029e100}, {0x788cf60, 0xc0005b0000})
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/app/cmd/build.go:71 +0x165
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.(*builder).NoArgs.func1(0xc00095b8c0, {0x5, 0x5, 0xc000a23698})
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/app/cmd/commands.go:138 +0x9a
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.applyPostRunHooks.func1(0xc000918840, {0xc000a001b0, 0x0, 0x3})
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/app/cmd/commands.go:185 +0x6e
github.com/spf13/cobra.(*Command).execute(0xc000918840, {0xc000a00180, 0x3, 0x3})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0xaf0
github.com/spf13/cobra.(*Command).ExecuteC(0xc000906c80)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x8b3
github.com/spf13/cobra.(*Command).ExecuteContextC(0xc000906c80, {0x78e13d0, 0xc00029e100})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:911 +0x8b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run({0x788bfc0, 0xc00012c008}, {0x788bfc0, 0xc00012c010})
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/app/skaffold.go:50 +0x4b1
main.main()
	/private/tmp/skaffold-20220625-10976-1ij4m21/cmd/skaffold/skaffold.go:49 +0x2bd
```
### Information

- Skaffold version: 1.39.0
- Operating system: Darwin AIH1013-TAILIFY.local 20.6.0 Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64 x86_64
- Installed via: Homebrew
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta27
kind: Config
metadata:
  name: fides
build:
  artifacts:
    - image: registry.tailify.com/tailify/fides-base
      context: .
      docker:
        dockerfile: k8s/docker/Dockerfile.base
      sync:
        infer:
          - /opt/*
    - image: registry.tailify.com/tailify/fides-code
      requires:
        - image: registry.tailify.com/tailify/fides-base
          alias: BASE
      sync:
        infer:
          - /opt/*
      context: .
      docker:
        dockerfile: k8s/docker/Dockerfile.code
    - image: registry.tailify.com/tailify/fides-spark
      context: .
      docker:
        dockerfile: k8s/docker/Dockerfile.spark
      requires:
        - image: registry.tailify.com/tailify/fides-code
          alias: BASE
    - image: registry.tailify.com/tailify/fides-backend
      context: .
      docker:
        dockerfile: k8s/docker/Dockerfile.backend
  platforms:
    - linux/amd64
  local:
    push: true
    useBuildkit: true
    tryImportMissing: true
  tagPolicy:
    inputDigest: {}
deploy:
  kustomize:
    buildArgs:
      - "--load-restrictor=LoadRestrictionsNone"
    paths:
      - k8s/base
```

### Steps to reproduce the behavior
TBD

**Comment by tejal29** [comment]
this has been fixed on v1.39.1

**Comment by akharlamov** [comment]
Unfortunately, that is not correct. Please, see the log below, it is from 1.39.1
```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.needsRemoteTagging.Tag({{0x14000bb6780, 0x40}, {0x14000eee1c0, 0x68}, {0x14000a78d20, 0x47}}, {0x107c03d18, 0x14000b96f30}, 0x1400001a800, {0x14000a91170})
	/private/tmp/skaffold-20220624-80169-j89lh6/pkg/skaffold/build/cache/details.go:94 +0x218
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.(*cache).Build(0x1400001a800, {0x107c03c70, 0x14000dc4180}, {0x107bb2580, 0x14000392690}, 0x14000b97110, {0x1400001ebe0, 0x4, 0x4}, {0x14000a91170}, ...)
	/private/tmp/skaffold-20220624-80169-j89lh6/pkg/skaffold/build/cache/retrieve.go:85 +0xd04
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*Builder).Build(0x140007c4600, {0x107c03d18, 0x14000b970b0}, {0x107bb2580, 0x14000392690}, {0x1400001ebe0, 0x4, 0x4})
	/private/tmp/skaffold-20220624-80169-j89lh6/pkg/skaffold/runner/build.go:105 +0x8fc
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doBuild.func1({0x107c4a820, 0x140007c4600}, {0x14000b704e0, 0x1, 0x1})
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/app/cmd/build.go:72 +0x118
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.withRunner({0x107c03c70, 0x14000030100}, {0x107bb2580, 0x14000a03310}, 0x140008e7530)
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/app/cmd/runner.go:55 +0x184
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doBuild({0x107c03c70, 0x14000030100}, {0x107bb2580, 0x14000a03310})
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/app/cmd/build.go:71 +0x11c
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.(*builder).NoArgs.func1(0x140003d0840, {0x140009d5a10, 0x0, 0x3})
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/app/cmd/commands.go:138 +0x9c
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.applyPostRunHooks.func1(0x140003d0840, {0x140009d5a10, 0x0, 0x3})
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/app/cmd/commands.go:185 +0x5c
github.com/spf13/cobra.(*Command).execute(0x140003d0840, {0x140009d59e0, 0x3, 0x3})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x998
github.com/spf13/cobra.(*Command).ExecuteC(0x140003c4c80)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x7b8
github.com/spf13/cobra.(*Command).ExecuteContextC(0x140003c4c80, {0x107c03c70, 0x14000030100})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:911 +0x74
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run({0x107bb15e0, 0x14000010018}, {0x107bb15e0, 0x14000010020})
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/app/skaffold.go:50 +0x3e4
main.main()
	/private/tmp/skaffold-20220624-80169-j89lh6/cmd/skaffold/skaffold.go:49 +0x29c

 🐍 fides-8TUr9fL4-py3.9  ⎈ |kubernetes-admin@gaia:dev2   fix/FID-XXX-docker-env-fides-dir
 ✘ aih1013@MacBook-Pro  ~/fides  skaffold version
v1.39.1
```
